tests/0126-oauthbearer_oidc.c

```diff
@@ -33,64 +33,140 @@
 
 
 /**
- * @brief After config OIDC fields, make sure the producer gets created
- *        successfully.
+ * @brief After config OIDC, make sure the producer and consumer
+ *        can work successfully.
  *
  */
-static void do_test_create_producer() {
+static void do_test_produce_consumer_with_OIDC(rd_kafka_conf_t *conf) {
+        if (test_needs_auth()) {
+                TEST_SKIP("Mock cluster does not support SSL/SASL\n");
+                return;
+        }
+
         const char *topic;
         uint64_t testid;
-        rd_kafka_t *rk;
-        rd_kafka_conf_t *conf;
-        rd_kafka_conf_res_t res;
-        char errstr[512];
+        rd_kafka_t *p1;
+        rd_kafka_t *c1;
+        int msgcounter;
+        rd_kafka_topic_t *rkt;
```

Comment:
Avoid using rkt in tests, it is a complex (and somewhat legacy) interface. There are test methods that take a topic string.